eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -817,7 +818,19 @@ instance (ShelleyEraImp era, Arbitrary a, Show a, Testable prop) => Example (a -
               (\f -> hook (\_st -> f ()))
               progressCallback
           void $ throwIO $ resultStatus res
-     in evaluateExample runImpTestExample params hook progressCallback
+     in adjustResult <$> evaluateExample runImpTestExample params hook progressCallback
+
+-- | Prevent HSpec from colorizing failures with all red color, which interferes with our
+-- custom coloring using Pretty.
+adjustResult :: H.Result -> H.Result
+adjustResult res = res {resultStatus = adjustResultStatus (resultStatus res)}
+  where
+    adjustResultStatus = \case
+      H.Failure mLoc reason -> H.Failure mLoc $ adjustFailureReason reason
+      resultStatus -> resultStatus
+    adjustFailureReason = \case
+      H.Reason reason -> H.ColorizedReason reason
+      failureReason -> failureReason
```

Comment:
I find this a bit more readable, but you might not:

```suggestion
      H.Failure mLoc (H.Reason reason) -> H.Failure mLoc (H.ColorizedReason reason)
      resultStatus -> resultStatus
```

Reply:
No longer applicable